vnext/proposals/active/keyboard-reconcile-desktop.md

```diff
@@ -0,0 +1,148 @@
+# Reconcile Keyboarding APIs in React Native - desktop
+
+There are several APIs and behaviors around keyboarding introduced in Windows and macOS, some of which match and some don&#39;t. Windows implementation is a bit more complete. The ultimate goal is to reconcile and come up with a singular set of keyboarding APIs and behaviors for RN desktop that can then be proposed for upstreaming into core as well as documented cleanly. This document captures the set of APIs and behaviors and identifies:
+
+- Deltas between implementations in order to reconcile one with the other
+- Deltas in scenarios that need to be completed in both
+- Any other fall out work
+
+## acceptsKeyboardFocus
+
+Property that determines whether this View, Touchable should be focusable with a non-touch input device, eg. receive focus with a hardware keyboard. Setting the property to true will make that View compoonent become a part of tabbing loop.
+
+| **TYPE** | **REQUIRED** | **PLATFORM** | **DEFAULT** |
+| --- | --- | --- | --- |
+| boolean | No | Windows, macOS | false |
+
+React Native core introduced `focusable` that does the same thing as acceptsKeyboardFocus in v0.62. In order to align and make this complete, the work here includes the following:
+- **COMPLETED** : Windows has added `focusable` and marked `acceptsKeyboardFocus` as deprecated in 0.62. 
+- macOS needs to do add `focusable` and mark `acceptsKeyboardFocus` as deprecated. Tracked by - [Issue#498](https://github.com/microsoft/react-native-macos/issues/498)
+- **COMPLETED** : Add focusable to Pressable ([windows Issue#5512](https://github.com/microsoft/react-native-windows/issues/5512), [macOS Issue#500](https://github.com/microsoft/react-native-macos/issues/500)). 
+- `focusable` is already supported in Android and exists in core. There is no hard upstreaming requirement, however – it would be good to implement this for iOS/iPadOS for completion.
+
+## focus, blur, onFocus, onBlur on View
+
+React Native core supports `.focus()`, `.blur()` methods and `onFocus()`, `onBlur()` callbacks on TextInput component. Windows and macOS implementations add the same methods and callbacks on View in addition to supporting them on TextInput. The work remaining for these include:
```

Comment:
I think we should try to follow W3C behavior that doesn't bubble blur/focus events. We don't intend for future focus-related APIs to retain the non-standard bubbling behavior in ReactDOM, and I suspect the same desire will hold true for RN.

Reply:
@acoates-ms , @FalseLobster, @tom-un   - can you comment on whether we bubble blur/focus events in RN windows/mac today?

Reply:
I think today we bubble on windows and do not on mac.  We were planning on reconciling those and making them both bubble  -- I see that on web there is another set of events, focusin/focusout, which is focus/blur but with bubbling.  Do we need to have both set of events in R/RN?

Reply:
I think we should have both sets of events in RN too. There are some really important places where you want focus/blur bubbling behavior, for example when dealing with focus management around dialogs/menus and handling focus restoration.

Reply:
Do you think the reason there are two events is for backward compat in web?  -- Did they start with onFocus/onBlur then realize that they messed up and should have made them bubble?  I'm trying to understand the reason for having both.  Consistency with web might be sufficient, but if we can simplify things a little that would also be great, since we will have to implement these events on at least 4 native platforms. -- (Could we just implement onFocusIn and onFocusOut?)

Reply:
I'm honestly not sure. That might be a good hunch though, as I've often wondered the same thing.

To note as well, ReactDOM's event system in v17 will switch to using `focusin`/`focusout` for `onFocus`/`onBlur` under the hood. This is because we want the native bubbling behavior (opposed to emulating it with `focus`/`blur` in the capture phase).

Reply:
Will ReactDOM's onFocus and onBlur bubble on the JS side too?

Reply:
focusin/out have different timing and semantics to focus/blur https://w3c.github.io/uievents/#event-type-focusin

Reply:
> Will ReactDOM's onFocus and onBlur bubble on the JS side too?

React's onFocus and onBlur have always bubbled, so we switched to use focusin/focusout to ensure this works for those events.

> focusin/out have different timing and semantics to focus/blur

What's interesting, and I found this out from working on JSDOM and making focusin/focusout work – it seems that no browsers follow the spec on how `focusin`/`focusout` works and instead all use the same workflow that differs. The spec says that `focusin` should fire before `focus` and `focusout` should happen before `blur`, but in reality they happen in reverse order: `focus`, `focusin`, `blur`, `focusout` and they even have another strange case when focusing through iframes.

Reply:
I've always figured we'd want both.  They feel like different use cases to me.  Our native platform has both concepts and they both see use.

I think in an ideal world we would follow the W3C naming conventions & behaviors, but this would require a breaking change for the onFocus & onBlur events in react-native today since those currently bubble.

Reply:
`focusin`, `focusout` semantically seem similar to Windows's `GettingFocus`, `LosingFocus`. Should be interesting to add. Have added a note to [Issue 5706](https://github.com/microsoft/react-native-windows/issues/5706) to tackle these in addition to `beforeBlur`, `afterBlur`.

Reply:
The landed PR https://github.com/microsoft/react-native-macos/pull/506 makes macOS onFocus/onBlur bubbling behave the same as react-native-windows.